code/modules/mining/machine_redemption.dm

```diff
@@ -149,10 +151,11 @@
 	))
 	signal.send_to_receivers()
 
-/obj/machinery/mineral/ore_redemption/process()
+/obj/machinery/mineral/ore_redemption/pickup_items()
 	if(!materials.mat_container || panel_open || !powered())
 		return
-	var/atom/input = get_step(src, input_dir)
+
+	var/atom/input = input_turf
 	var/obj/structure/ore_box/OB = locate() in input
```

Comment:
Not a fan of this, you're looping through the entire Turf when you can just get the ore that was dropped from this procs params

Reply:
yeah I will make all pickup_item procs across all these machines not loop through the turf and instead just suck up the 1 item that entered.

Reply:
👌 awesome